src/main/java/com/github/manolo8/darkbot/core/objects/facades/DiminishQuestMediator.java

```diff
@@ -0,0 +1,59 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.gui.DiminishQuestGui;
+import eu.darkbot.api.API;
+import eu.darkbot.api.managers.QuestAPI;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+public class DiminishQuestMediator extends Updatable implements API.Singleton {
+    private State state = State.NOT_EXIST;
+    private final DiminishQuestGui diminishQuestGui;
+    private final QuestProxy.Quest diminishQuest = new QuestProxy.Quest();
+
+    public DiminishQuestMediator(DiminishQuestGui diminishQuestGui) {
+        this.diminishQuestGui = diminishQuestGui;
+    }
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            state = State.NOT_EXIST;
+        } else if (!"diminish_quests".equals(API.readString(address, 0x38))) {
+            address = 0;
+            state = State.NOT_EXIST;
+        } else {
+            state = State.NOT_EXIST;
+            int stateIndex = API.readInt(address, 0x50, 0x40) + 1;
+            if (stateIndex >= 0 && stateIndex < 5) {
+                state = State.values()[stateIndex];
+                diminishQuest.update(API.readAtom(address, 0x50, 0x60));
+            }
+        }
```

Comment:
```suggestion
        if (!isValid()) {
            state = State.NOT_EXIST;
            return;
        }
        int stateIndex = API.readInt(address, 0x50, 0x40) + 1;
        if (stateIndex >= 0 && stateIndex < STATES.length) {
            state = STATES[stateIndex];
            diminishQuest.update(API.readAtom(address, 0x50, 0x60));
        } else {
            state = State.NOT_EXIST;
        }
```
i'd simplify the method in this logic, move validation check (addr != 0, and the other string check) into its own method, and make sure to check against the actual size of states instead of hard-coded `5`
also just add a `private static final State[] STATES = State.values()` to avoid the allocations every tick.

Also, what's the +1? is there an extra state 0 in-game we're ignoring here?

Reply:
Possible state indexes: 

- -1 when quest not active (timer left etc), 
- 0 when waiting accepting, 
- 2 when quest accepted and active. 
- 3 when quest done.

'+1' is mapping in-game index to enum index.